components/raftstore/src/store/peer_storage.rs

```diff
@@ -757,7 +797,105 @@ where
         Ok(())
     }
 
-    pub fn entries(&self, low: u64, high: u64, max_size: u64) -> raft::Result<Vec<Entry>> {
+    pub fn set_async_fetch_res(&mut self, res: RaftlogFetchResult) {
+        let prev_res = self.async_fetch_res.replace(Some(res));
```

Comment:
Technically, it should validate whether the logs still match the actual logs. For example, a leader fetches logs between [10, 50) asynchronously. And then it becomes follower, and logs are overwritten by new leader from 40. And then it becomes a leader again. If the returned value is reused, then it may produce incorrect logs sequence.

An easy way to solve the problem is store the term when trigger the task, and check whether term changes when the task is returned. Though this may be a waste to drop all logs if they are not overwritten at all.

Reply:
seems we should add a term field in GetEntriesGetContext? peer storage has no access to the current term